src/operator/tensor/elemwise_binary_broadcast_op.h

```diff
@@ -130,20 +101,19 @@ void BinaryBroadcastCompute(const nnvm::NodeAttrs& attrs,
                             const std::vector<TBlob>& inputs,
                             const std::vector<OpReqType>& req,
                             const std::vector<TBlob>& outputs) {
+  using namespace broadcast;
   TShape new_lshape, new_rshape, new_oshape;
-  bool need_bc = BinaryBroadcastShapeCompact(inputs[0].shape_, inputs[1].shape_, outputs[0].shape_,
-                                             &new_lshape, &new_rshape, &new_oshape);
-  if (!need_bc) {
+  int ndim = BinaryBroadcastShapeCompact(inputs[0].shape_, inputs[1].shape_, outputs[0].shape_,
+                                         &new_lshape, &new_rshape, &new_oshape);
+  if (!ndim) {
     BinaryCompute<xpu, OP>(attrs, ctx, inputs, req, outputs);
   } else {
+    mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
     MSHADOW_TYPE_SWITCH(outputs[0].type_flag_, DType, {
-      if (new_oshape.ndim() == 2) {
-        BinaryBroadcastComputeImpl<xpu, 2, DType, OP>(
-          ctx, inputs, req, outputs, new_lshape, new_rshape, new_oshape);
-      } else {
-        BinaryBroadcastComputeImpl<xpu, MXNET_SPECIAL_MAX_NDIM, DType, OP>(
-          ctx, inputs, req, outputs, new_lshape, new_rshape, new_oshape);
-      }
+      CTensor<DType> lhs = inputs[0].get_with_shape<xpu, MAX_DIM, DType>(new_lshape.get<MAX_DIM>(), s);
```

Comment:
You can directly pass TBlob since you are just using dptr_ and not using any of the tensor features.